nc_time_axis/__init__.py

```diff
@@ -233,14 +248,20 @@ def convert(cls, value, unit, axis):
                 return value
             first_value = value
 
-        if not hasattr(first_value, 'calendar'):
-            raise ValueError('A "calendar" attribute must be attached to '
-                             'netcdftime object to understand them properly.')
-        return netcdftime.date2num(value, cls.standard_unit,
-                                   calendar=first_value.calendar)
+        if not isinstance(first_value.datetime, netcdftime.datetime):
+            raise ValueError('The datetime attribute of the CalendarDateTime '
+                             'object must be of type `netcdftime.datetime`.')
+
+        ut = netcdftime.utime(cls.standard_unit, calendar=first_value.calendar)
+
+        if isinstance(value, Iterable):
+            num = ut.date2num([v.datetime for v in value])
+        else:
+            num = ut.date2num(value.datetime)
```

Comment:
@lbdreyer It might be that this is an artificial case (for a testing perspective) and when plugged into matplotlib we'd never see this. Perhaps you could confirm and we'll take it from there.